### ISSUE TITLE
auditTime(0, animationFrameScheduler) stops emiting unexpectedly.

### ISSUE THREAD

**Issue by fdeslandes-wk** [ReactiveX/rxjs#7096]
### Describe the bug

Since version 7.5.0, there is a weird interaction between different observables using animationFrameScheduler, especially auditTime(0, animationFrameScheduler), which stops emitting when other observables using the animationFrameScheduler started concludes after the first event is sent to the audit.

This causes problems with Angular CDK virtual scroll viewport which stops working when other things in the application use the animationFrameScheduler on observables which concludes.

It seems related to the same changes which caused https://github.com/ReactiveX/rxjs/issues/7017.

### Expected behavior

Other observables should not have side effects on the animationFrameScheduler Action from auditTime(0, animationFrameScheduler) and auditTime should keep emitting regardless of the recycling of animationFrameScheduler actions.

### Reproduction code

```typescript
See the StackBlitz code, which reproduces the issue.  The issue cannot be reproduced using rxJs 7.4.x. It is there since 7.5.x.
```

### Reproduction URL

https://rxjs-afsbug-s5ubbu.stackblitz.io

### Version

7.5.0 - 7.5.7

**Comment by gabrielgagne** [comment]
Same problem here

**Comment by endoffline** [comment]
We also have the same problem. In my case it leads to the Angular CDK virtual scroll viewport to not render the necessary elements which would be visible within the current viewport, as scroll events are not processed any longer.

**Comment by gabrielgagne** [comment]
Yes, I noticed this bug too after seeing cdk virtual scroll (and our other component relying on animationFrameScheduler) being messed up after updating past 7.4

**Comment by endoffline** [comment]
I've expanded the example @fdeslandes-wk and included the `cdk-virtual-scroll-viewport` to highlight the problem more prominently.

Have a look here: https://stackblitz.com/edit/angular-zdbuly?file=src%2Fmain.ts

**Comment by medi6** [comment]
Hi, I've got the same problem here...

**Comment by ChristophWieske** [comment]
Same here with 7.8.1

Is there a workaround?

**Comment by rubber-goose** [comment]
I am not able to reproduce the issue with 7.8.2 in the stackblitz provided by @endoffline 

Maybe it was fixed with: #7444

**Comment by endoffline** [comment]
Hey, I've taken a look on my example using rxjs@7.8.2 and I can confirm that the issue doesn't seem to appear anymore. Looks fine to me now.

Thanks for pining me about this.
